kubric/core/base.py

```diff
@@ -76,16 +76,19 @@ def active_scene(self):
 
   @tl.default("uid")
   def _uid(self):
-    """ Use the class name and a global count separated by a period as the UID.
-    Counting starts at 1 and always has at least three digits.
-    UIDs are thus of the form Scene.001 or Cube.004.
-    That way the first 999 instances of each class can be sorted alphabetically.
-    """
-    name = self.__class__.__name__
+    # e.g. if self.name="Cube", the UIDs of the first three: {"Cube", "Cube.001", "Cube.002"}
+    # Matches blender naming logic, and allows lexicographical sorting of the first 999 instances.
+
+    # Undefined assets (e.g. UndefinedMaterial) are singletons and do not have a name
```

Comment:
Misleading: They do have a name. Otherwise you couldn't return `self.name`. Change to the following? 
> Undefined assets (e.g. UndefinedMaterial) are singletons and thus their uid is always equal to their name